pkg/wallet/wallet.go

```diff
@@ -521,18 +546,82 @@ func AddressIsUnique(address string) error {
 	return nil
 }
 
+// Helpers: public key
+
 // GetPupKey returns the public key of the wallet.
 func (wallet *Wallet) GetPupKey() string {
-	return PublicKeyPrefix + base58.CheckEncode(wallet.KeyPair.PublicKey, Base58Version)
+	return PublicKeyPrefix + base58.CheckEncode(RemovePubKeyVersion(wallet.KeyPair.PublicKey), PubKeyVersion)
+}
+
+func addPubKeyVersion(pubKey []byte) []byte {
+	return append([]byte{PubKeyVersion}, pubKey...)
+}
+
+// RemovePubKeyVersion removes the version byte of a public key if the public key has a version byte.
+func RemovePubKeyVersion(versionedPubKey []byte) []byte {
+	return versionedPubKey[1:]
+}
+
+// CheckPukKeyVersion checks the version byte of a public key.
+// Return an error if the version byte is unknown.
+// Return the public key with the version byte otherwise.
+func CheckPukKeyVersion(versionedPubKey []byte) ([]byte, error) {
+	// Declare here all known public key versions.
+	knownVersions := []byte{PubKeyVersion}
+
+	for _, version := range knownVersions {
+		if versionedPubKey[0] == version {
+			return versionedPubKey, nil
+		}
+	}
+
+	return nil, fmt.Errorf("invalid version byte")
 }
```

Comment:
You should create a VersionKey object with different methods to improve code clarity and avoid duplicating the code for public keys and private keys.

Something like:
```golang
type VersionedKey byte[]

func (v VersionKey) AddVersion(version int) []byte {
  return append([]byte{version}, v...)
}

func (v VersionKey) RemoveVersion() []byte {
  return v[1:]
}

...
```

Reply:
done in https://github.com/massalabs/station-massa-wallet/pull/530/commits/576f72db2bd3d20acb805b4ebb49eb47692138fc